cloud/shared/bin/lib/terraform.py

```diff
@@ -45,16 +103,42 @@ def perform_apply(
         print(" - Test. Not applying terraform.")
         return True
 
-    print(" - Run terraform apply")
     # Enable compact-warnings as we have a bunch of
     # "value of undeclared variables" warnings as some variables used in one
     # deployment (e.g. aws) but not the other.
-    terraform_apply_cmd = f'{terraform_cmd} apply -input=false -var-file={tf_vars_filename} -compact-warnings'
+    terraform_apply_cmd = f'terraform -chdir={terraform_template_dir} apply -input=false -var-file={tf_vars_filename} -compact-warnings'
     if config_loader.skip_confirmations:
         terraform_apply_cmd += ' -auto-approve'
     if is_destroy:
         terraform_apply_cmd += ' -destroy'
-    subprocess.check_call(shlex.split(terraform_apply_cmd))
+
+    print(f" - Run {terraform_apply_cmd}")
+
+    output, exit_code = capture_stderr(terraform_apply_cmd)
+    if exit_code:
+        if "Error acquiring the state lock" in output:
+            # Lock ID is a standard UUID v4 in the form 00000000-0000-0000-0000-000000000000
+            match = re.search(
+                    r'ID:\s+([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})',
+                    output)
+            error_text = inspect.cleandoc(
+                """
+                                     The Terraform state lock can not be acquired.
+                                     This can happen if you are running a command in another process, or if another Terraform process exited prematurely. 
+                                     """)
+            if match:
```

Comment:
**thought:** If there is a match, is there a way of raising an error and prompting the user if they want to force unlock with the matched ID and if they do we do it for them without them having to re-run? 

Just thinking that would allow civic entities to proceed without needing to update civiform-deploy hypothetically, so may be good to have if possible.

Reply:
Got it, yeah that makes sense! I think giving the option of doing it for them if they're running interactively already is nice and printing the message if not SGTM. I'll write in the thread as well.